pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -297,24 +297,15 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 							con.mu.Unlock()
 							continue
 						}
-					} else if discReq.ErrorDetail != nil {
-						// If versions mismatch then we should either have an error detail or no routes if a protocol error has occurred
-						if discReq.ErrorDetail != nil {
-							errCode := codes.Code(discReq.ErrorDetail.Code)
-							adsLog.Warnf("ADS:RDS: ACK ERROR %v %s %s:%s", peerAddr, con.ConID, errCode.String(), discReq.ErrorDetail.GetMessage())
-							incrementXDSRejects(rdsReject, con.node.ID, errCode.String())
-						}
-						continue
 					} else if len(routes) == 0 {
 						// XDS protocol indicates an empty request means to send all route information
 						// In practice we can just skip this request, as this seems to happen when
 						// we don't have any routes to send.
 						continue
 					}
 				}
-
 				con.Routes = routes
-				adsLog.Debugf("ADS:RDS: REQ %s %s routes:%d", peerAddr, con.ConID, len(con.Routes))
+				adsLog.Infof("ADS:RDS: REQ %s %s routes:%d", peerAddr, con.ConID, len(con.Routes))
```

Comment:
Noisy

Reply:
RDS REQ is sent only when necessary. Not everytime envoy received LDS.

Reply:
> RDS REQ is sent only when necessary. Not every time envoy received LDS.

Every time an LDS is sent,  Envoy sends RDS for the routes referred by the listeners - In which case does it skip? I can think of only case where route config is given inline. 

IMO,  debug is the right level for these kind of messages.

Reply:
OK, will restore